src/main/java/org/apache/sysds/runtime/instructions/fed/QuaternaryWCeMMFEDInstruction.java

```diff
@@ -59,17 +60,17 @@ public void processInstruction(ExecutionContext ec)
 		MatrixObject U = ec.getMatrixObject(input2);
 		MatrixObject V = ec.getMatrixObject(input3);
 		ScalarObject eps = null;
-		
+
 		if(qop.hasFourInputs()) {
 			eps = (_input4.getDataType() == DataType.SCALAR) ?
 				ec.getScalarInput(_input4) :
 				new DoubleObject(ec.getMatrixInput(_input4.getName()).quickGetValue(0, 0));
 		}
 
-		if(!(X.isFederated() && !U.isFederated() && !V.isFederated()))
+		if(!(X.isFederated(FType.ROW) && !U.isFederated() && !V.isFederated()))
```

Comment:
Could we convert this to a positive case instead? Something more like: 
`if (X.isFederated(FType.ROW) && !U.isFederated() && !V.isFederated()){
//process instruction
}
else {
//throw exception
}`
I think this would be easier to read and leave the method open for new cases in the future, for instance to support col federation. 
This comment also applies to QuaternaryWSLossFEDInstruction and QuaternaryWSigmoidFEDInstruction.

Reply:
I converted it to the positive case for all five federated quaternary operations.